File: Memo/howsum.js
const howsum = (target,numbers,memo= {}) => {
    if(target in memo) return memo[target];
    if(target===0) return [];
    if(target < 0) return null;

    for(let num of numbers){
        const rem=target-num;
        result={}
        result= howsum(rem,numbers,memo);
        if(result !== null){
            memo[target]=[...result,num];
            return memo[target];
        }
    }
    memo[target]=null;
    return null;
}

console.log(howsum(100,[3,1,2,3]));

File: Memo/howsum.cpp
#include<bits/stdc++.h>
 
using namespace std; 
  
vector<int> howsum(int target,vector<int> &nums)
{
    if(target==0) return {};
    if(target<0) return ;

    for(auto x:nums){
        int rem=target-x;
        vector<int> result=howsum(rem,nums);
        if(!result.empty())
        {
            result.push_back(x);
            return result;
        }
    }
    return {};
}

int main()
{
    vector<int> k={2,3};
    vector<int> res=howsum(8,k);
    for(auto x:res)
        cout<<x<<endl;
    return 0;
}